l10n.py

```diff
@@ -48,235 +63,325 @@
 
-    def file(self, lang, plugin_path=None):
+        return pathlib.Path(LOCALISATION_DIR)
+
+    def file(self, lang: str, plugin_path: Optional[str] = None) -> Optional[TextIO]:
+        """
+        Open the given lang file for reading.
+
+        :param lang: The lang file to open (just the name of the lang)
+        :param plugin_path: path to plugins dir, to check for plugin based lang files, defaults to None
+        :return: the opened file (Note: This should be closed when done)
+        """
         if plugin_path:
-            f = join(plugin_path, '%s.strings' % lang)
+            f = join(plugin_path, f'{lang}.strings')
             if exists(f):
                 try:
-                    return codecs.open(f, 'r', 'utf-8')
-                except:
-                    print_exc()
+                    return open(f, 'r', encoding='utf-8')
+
+                except Exception:
+                    logger.exception(f'could not open {f}')
+
             return None
-        elif getattr(sys, 'frozen', False) and platform=='darwin':
-            return codecs.open(join(self.respath(), '%s.lproj' % lang, 'Localizable.strings'), 'r', 'utf-16')
-        else:
-            return codecs.open(join(self.respath(), '%s.strings' % lang), 'r', 'utf-8')
 
+        elif getattr(sys, 'frozen', False) and platform == 'darwin':
+            return (self.respath() / f'{lang}.lproj' / 'Localizable.strings').open('r', encoding='utf-16')
 
-class Locale(object):
+        return (self.respath() / f'{lang}.strings').open('r', encoding='utf-8')
 
-    def __init__(self):
-        if platform=='darwin':
+
+class _Locale:
+    """Locale holds a few utility methods to convert data to and from localized versions."""
+
+    def __init__(self) -> None:
+        if platform == 'darwin':
             self.int_formatter = NSNumberFormatter.alloc().init()
             self.int_formatter.setNumberStyle_(NSNumberFormatterDecimalStyle)
             self.float_formatter = NSNumberFormatter.alloc().init()
             self.float_formatter.setNumberStyle_(NSNumberFormatterDecimalStyle)
             self.float_formatter.setMinimumFractionDigits_(5)
             self.float_formatter.setMaximumFractionDigits_(5)
 
-    def stringFromNumber(self, number, decimals=None):
-        # Uses the current system locale, irrespective of language choice.
-        # Unless `decimals` is specified, the number will be formatted with 5 decimal
-        # places if the input is a float, or none if the input is an int.
+    def stringFromNumber(self, number: Union[float, int], decimals: int = None) -> str:  # noqa: N802
+        warnings.warn(DeprecationWarning('use _Locale.string_from_number instead.'))
+        return self.stringFromNumber(number, decimals)
+
+    def numberFromString(self, string: str) -> Union[int, float, None]:  # noqa: N802
+        warnings.warn(DeprecationWarning('use _Locale.number_from_string instead.'))
+        return self.numberFromString(string)
```

Comment:
Again with what looks like infinite recursion to me.

Reply:
This and the other one have been fixed